### PR TITLE
Cleaning (?) headers

### PR DESCRIPTION
**Pull request by Iximiel**
##### Description

I found some header shortcuts here and there, mainly to ActionRegister.h (but also to CLToolRegister.h)
I removed them and I changed the include directives where is needed to make the compilation possible.

It compiles, I expect the regtests to pass, so I'm starting the PR to control that the CI will run smoothly

I have a question about CLtool.* and CLToolRegister.*, should not them be in cltools? (And cltools can be included both from core (in cltools in incuded from there) and from cltools (in exernal modules is included from there), I think that is confusing)

Since the nature of these changes I think this PR should squashed and merged instead of a simple merge

##### Target release

I would like my code to appear in the next release

##### Type of contribution

- [X] changes to code or doc authored by PLUMED developers, or additions of code in the core or within the default modules
- [X] changes to a module not authored by you
- [ ] new module contribution or edit of a module authored by you

##### Copyright

- [X] I agree to transfer the copyright of the code I have written to the PLUMED developers or to the author of the code I am modifying.

- [ ] the module I added or modified contains a `COPYRIGHT` file with the correct license information. Code should be released under an open source license. I also used the command `cd src && ./header.sh mymodulename` in order to make sure the headers of the module are correct. 

##### Tests

- [ ] I added a new regtest or modified an existing regtest to validate my changes.
- [x] I verified that all regtests are passed successfully on [GitHub Actions](https://github.com/plumed/plumed2/actions).

### CODE REVIEW

**Comment by GiovanniBussi** [comment]
Regarding cltools, the cltools module can be switched off (default-on) whereas the core module is compulsory, so it cannot depend on it.

**Comment by GiovanniBussi** [comment]
Also please check the diff, I can see many unrelated changes including some bulky change in pathtools and isdb. Are these included by mistake or are they related to this cleanup?

**Comment by Iximiel** [comment]
I fear that most of the changes in the two files with a lot of changes are slipped in by error, by seeing ho diff is confused (i need to deactivate the auto formatter)

File: src/ves/MD_LinearExpansionPES.cpp
Comment (GiovanniBussi) [inline src/ves/MD_LinearExpansionPES.cpp:1]: This is still heavily modified, could you please revert and only add the relevant change?

File: .gitignore
[the remark this answers is not in the document]
Comment (Iximiel) [reply .gitignore:1]: I found a workaround for stopping me to keep pushing .vscode and build to the gitignore

**Pull request merged**